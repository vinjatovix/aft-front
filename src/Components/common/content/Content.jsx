import React from "react";
import PropTypes from "prop-types";
import { ActionButton } from "../../ui/buttons/ActionButton";
import { modalsStateType } from "../../../PropTypes";

export const Content = ({ children, modals, actions, isEditor, isOpen }) => {
  return (
    <div className={modals.blur ? "content blur" : "content"}>
      <div className="content_header">
        {(isEditor || isOpen) && (
          <ActionButton text="AÃ±adir" type="bg-green" action={actions.add} />
        )}
      </div>
      {children}
    </div>
  );
};

Content.propTypes = {
  children: PropTypes.node.isRequired,
  modals: modalsStateType,
  actions: PropTypes.shape({
    add: PropTypes.func.isRequired,
  }).isRequired,
  isEditor: PropTypes.bool,
};

Content.defaultProps = {
  isEditor: false,
};
