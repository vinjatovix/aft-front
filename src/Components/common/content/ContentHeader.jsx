import React from "react";
import PropTypes from "prop-types";

export const ContentHeader = ({ book, title, count }) => {
  return (
    <>
      {book && (
        <>
          <img
            className="portrait"
            src={
              book.img
                ? book.img
                : "https://boxshot.com/3d-book-cover/how-to-make-a-3d-book-cover-in-photoshop/sample.jpg"
            }
            alt="book"
          />
          <h1>
            {book.author}-{book.name}
          </h1>
          <h2>{book.description}</h2>
        </>
      )}

      <h3 data-testid={title}>
        {title} {count}
      </h3>
    </>
  );
};

ContentHeader.propTypes = {
  book: PropTypes.shape({
    name: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    img: PropTypes.string,
    description: PropTypes.string.isRequired,
  }),
  title: PropTypes.string.isRequired,
  count: PropTypes.number,
};

ContentHeader.defaultProps = {
  book: null,
  count: null,
};
