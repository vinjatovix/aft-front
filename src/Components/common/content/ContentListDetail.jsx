import React from "react";
import PropTypes from "prop-types";
import { AdminButtons } from "../../ui/admin/AdminButtons";

export const ContentListDetail = ({ actions, item, token, isAdmin, type }) => {
  const scene = (
    <p>
      {item.name}-{item.location} {item.time}: {item.description}.
    </p>
  );
  const character = (
    <p>
      {item.name}: {item.description}. {item.center}
    </p>
  );

  return (
    <>
      <div
        onClick={() => {
          actions.setDataDetail(item);
        }}
      >
        {type === "scene" && scene}
        {type === "character" && character}
      </div>
      {isAdmin && (
        <AdminButtons type="row" actions={actions} item={item} token={token} />
      )}
    </>
  );
};

ContentListDetail.propTypes = {
  type: PropTypes.string.isRequired,
  actions: PropTypes.shape({
    setDataDetail: PropTypes.func.isRequired,
    add: PropTypes.func.isRequired,
    close: PropTypes.func.isRequired,
    delete: PropTypes.func.isRequired,
    edit: PropTypes.func.isRequired,
    detail: PropTypes.func.isRequired,
    refresh: PropTypes.func.isRequired,
  }).isRequired,
  item: PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    _id: PropTypes.string.isRequired,
  }).isRequired,
  token: PropTypes.string.isRequired,
  isAdmin: PropTypes.bool,
};

ContentListDetail.defaultProps = {
  isAdmin: false,
};
