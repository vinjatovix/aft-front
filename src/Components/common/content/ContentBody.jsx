import React from "react";
import PropTypes from "prop-types";
import { ListItem } from "../ListItem";
import { ContentListDetail } from "./ContentListDetail";

export const ContentBody = ({ content, isAdmin, actions, token, type }) => {
  return (
    <ul className="content-list">
      {content &&
        content.map((item) => (
          <ListItem key={item._id} className="content-list_item" {...item}>
            <ContentListDetail
              type={type}
              isAdmin={isAdmin}
              token={token}
              actions={actions}
              item={item}
            />
          </ListItem>
        ))}
    </ul>
  );
};

ContentBody.propTypes = {
  content: PropTypes.array,
  isAdmin: PropTypes.bool,
  actions: PropTypes.object.isRequired,
  token: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};

ContentBody.defaultProps = {
  content: [],
  isAdmin: false,
};
