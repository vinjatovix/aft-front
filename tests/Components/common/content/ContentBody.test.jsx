import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { ContentBody } from "../../../../src/Components/common/content/ContentBody";
import mockActions from "../../../fixtures/mockActions";

describe("ContentBody Component", () => {
  it("should render scenes content with admin buttons", async () => {
    const { container } = render(
      <Router>
        <ContentBody
          type="scenes"
          actions={mockActions}
          token="token"
          isEditor={true}
          content={[
            {
              _id: "5e9f8f8f8f8f8f8f8f8f8f8",
              name: "Escena",
              description: "Descripción",
              metadata: {
                createdBy: "user",
              },
              character: {
                book: {
                  name: "Obra",
                },
                name: "Personaje",
              },
              scene: {
                name: "Escena",
              },
            },
            {
              _id: "5e9f8f8f8f8f8f8f8f8f8f9",
              name: "Escena",
              description: "Descripción",
              metadata: {
                createdBy: "user",
              },
              character: {
                book: {
                  name: "Obra",
                },
                name: "Personaje",
              },
              scene: {
                name: "Escena",
              },
            },
          ]}
        />
      </Router>
    );

    expect(container).toMatchSnapshot();
    expect(screen.getAllByTestId("action-button")).toHaveLength(4);
  });

  it("should render characters content without admin buttons", async () => {
    const { container } = render(
      <Router>
        <ContentBody
          type="characters"
          actions={mockActions}
          token="token"
          isEditor={false}
          content={[
            {
              _id: "6301221d03b6fbb712bbde7f",
              name: "Agnes Kris",
              book: {
                _id: "6301216103b6fbb712bbde7c",
                name: "transmitter Refined",
                author: "Theresa Larkin",
                description:
                  "I'll index the auxiliary SAS program, that should interface the USB driver!",
                metadata: {
                  createdAt: "2022-08-20T18:01:05.244Z",
                  createdBy: "userdev",
                  updatedAt: "2022-08-20T18:01:05.244Z",
                  updatedBy: "userdev",
                },
              },
              center: "instintive",
              description: "encompassing",
              metadata: {
                createdAt: "2022-08-20T18:04:13.283Z",
                createdBy: "userdev",
                updatedAt: "2022-08-20T18:04:13.283Z",
                updatedBy: "userdev",
              },
            },
            {
              _id: "6301221d03b6fbb712bbde7s",
              name: "Agnes KROS",
              book: {
                _id: "6301216103b6fbb712bbde7c",
                name: "transmitter RefinedOASI",
                author: "Theresa Larkin",
                description:
                  "I'll index the auxiliary SAS program, that should interface the USB driver!",
                metadata: {
                  createdAt: "2022-08-20T18:01:05.244Z",
                  createdBy: "userdev",
                  updatedAt: "2022-08-20T18:01:05.244Z",
                  updatedBy: "userdev",
                },
              },
              center: "instintive",
              description: "encompassing",
              metadata: {
                createdAt: "2022-08-20T18:04:13.283Z",
                createdBy: "userdev",
                updatedAt: "2022-08-20T18:04:13.283Z",
                updatedBy: "userdev",
              },
            },
          ]}
        />
      </Router>
    );

    expect(container).toMatchSnapshot();
    expect(screen.queryByTestId("action-button")).not.toBeInTheDocument();
  });
});
