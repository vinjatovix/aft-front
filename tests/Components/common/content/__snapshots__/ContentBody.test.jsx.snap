// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBody Component should render characters content without admin buttons 1`] = `
<div>
  <ul
    class="content-list"
  >
    <li
      class="content-list_item"
      data-testid="Agnes Kris"
    >
      <div />
    </li>
    <li
      class="content-list_item"
      data-testid="Agnes KROS"
    >
      <div />
    </li>
  </ul>
</div>
`;

exports[`ContentBody Component should render scenes content with admin buttons 1`] = `
<div>
  <ul
    class="content-list"
  >
    <li
      class="content-list_item"
      data-testid="Escena"
    >
      <div />
      <div
        class="control-panel-mini row"
      >
        <button
          class="btn-big bg-yellow"
          data-testid="action-button"
        >
          📝
        </button>
        <button
          class="btn-big bg-red"
          data-testid="action-button"
        >
          🗑️
        </button>
      </div>
    </li>
    <li
      class="content-list_item"
      data-testid="Escena"
    >
      <div />
      <div
        class="control-panel-mini row"
      >
        <button
          class="btn-big bg-yellow"
          data-testid="action-button"
        >
          📝
        </button>
        <button
          class="btn-big bg-red"
          data-testid="action-button"
        >
          🗑️
        </button>
      </div>
    </li>
  </ul>
</div>
`;
